crates/guest/src/guest.rs

```diff
@@ -159,26 +160,3 @@ macro_rules! try_ptr {
         }
     }};
 }
-
-#[cfg(test)]
-pub mod tests {
```

Comment:
These tests were never being run.

If they are worth keeping, should they be moved into the `host` crate and only run when the feature flag `error_as_host` is not enabled?

Reply:
The tests seem reasonable enough to me, I probably would try to keep them.

If we always use the feature flag though, I don't see a reason not to remove that?

Reply:
Ah okay I was confused. This is about the default behavior of passing a string into `wasm_error!()`.

By default this will produce a `WasmErrorInner::Guest("my error")`, in `hdi` it follows this behavior. When the `error_as_host` feature is enabled, it will produce a `WasmErrorInner:Host("my error")`, in `holochain` and `holochain_wasmer_host` it follows this behavior.

Because the `wasm_error!()` fn and its default behavior is actually in `crates/common`, I moved both this test and the mirroring test (in `crates/host/src/guest.rs`) into `crates/common` `crates/common/src/result.rs`, where wasm_error is defined. I put each test behind the feature flags required for their behavior. Let me know if that looks good.